src/mono/mono/mini/interp/transform.c

```diff
@@ -5321,7 +5321,7 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 
 				// optimize "box + ldnull + ceq/cgt_un" to true/false
 				// e.g. `t == null` in generic code when t is a vt/primitive (but not nullable)
-				if (*(td->ip + 5) == CEE_LDNULL) {
+				if (td->sp + 7 >= td->stack && *(td->ip + 5) == CEE_LDNULL) {
 					guint8 ceq = *(td->ip + 7);
```

Comment:
This doesn't look good.

Reply:
@vargaz any thoughts how to improve it? I am not yet familiar with interpreter internals 🙁

Reply:
Need to check ip+ 7 against the end of the code.

Reply:
I.e. td->ip - td->il_code + 7 < td->code_size

Reply:
thanks!